File: 初级算法/字符串/最长公共前缀.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty())
            return "";
        vector<string>::iterator str_pos = strs.begin();
        int size =  2147483647;
        string result;
        char temp;
        for(int i = 0; i < strs.size(); i ++){
            if(size > (*str_pos).size())
                size = (*str_pos).size();
            ++ str_pos;
        }
        str_pos = strs.begin();
        for(int i = 0;i < size;++ i){
            temp = *((*str_pos).begin() + i);    //取vector中第1个字符串中的第i个字符
            while(str_pos != strs.end()){
                if(temp != *((*str_pos).begin() + i))
                    return result;
                ++ str_pos;
            }
            result.append(1,temp);
            str_pos = strs.begin();
        }
        return result;
    }
};

File: java/最长公共前缀.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String result = "";
        String temp = "";
        if(strs.length == 0)
            return result;
//        if(strs.length == 1)
//            return strs[0];
        for(int i = 1; i < strs[0].length() + 1 ; i++){
            temp = strs[0].substring(0,i);
            for(int j = 0 ; j < strs.length ; j++){
                if(strs[j].startsWith(temp))
                    continue;
                else 
                    return result;
            }
            result = temp;
        }
        return result;
    }
}